js/src/api/contract/contract.js

```diff
@@ -269,6 +217,11 @@ export default class Contract {
       }
     }
 
+    return this._subscribe(event, options, callback);
+  }
+
+  _subscribe (event = null, _options, callback) {
+    const subscriptionId = this._subscriptions.length;
```

Comment:
Right, I haven't noticed that the array is not cleared. I guess separate counter and object for subscriptions could work the same way (you would iterate over `Object.values`)